src/modules/job-exec/bulk-exec.c

```diff
@@ -264,7 +264,8 @@ static struct exec_cmd *exec_cmd_create (const struct idset *ranks,
         fprintf (stderr, "exec_cmd_create: flux_cmd_copy failed");
         goto err;
     }
-    c->flags = flags;
+    /* bulk-exec defaults to use unbuffered for performance */
+    c->flags = flags | FLUX_SUBPROCESS_FLAGS_LOCAL_UNBUF;
```

Comment:
I'd just say "bulk-exec always uses unbuffered reads for performance" since calling it a default sort of implies that there could be a way to change it.